### ISSUE TITLE
`padded-blocks`: Add "start" & "end" options

### ISSUE THREAD

**Issue by erosman** [eslint-stylistic/eslint-stylistic#308]
### Add "start" & "end" options

It would be useful to have more control over the "padded-blocks".
_(I write classes/switches with padded start only.)_

```js
/* eslint @stylistic/js/padded-blocks: ["error", { "blocks": "never", "classes": "start" }] */

if (a) {
  b();
}

class C {

  static {
    a();
  }
}

```

### Suggested solution

Add "start" & "end" options to have more control over padding blocks

- "start": always at start, never at end
- "end": never at start, always at end

The rule already differentiates between the start & end padding i.e.

https://github.com/eslint-stylistic/eslint-stylistic/blob/beef00fed4e0ba46440996e1b7fceb364a769538/packages/eslint-plugin-js/rules/padded-blocks/padded-blocks.ts#L193-L194

### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [ ] If this feature request is accepted, I am willing to submit a PR to fix this issue

**Comment by antfu** [comment]
Sounds good to me, PR welcome!

**Comment by erosman** [comment]
Updated code in https://github.com/eslint-stylistic/eslint-stylistic/issues/308#issuecomment-2029481957

**Comment by 9romise** [comment]
I'm willing to make a PR for this. Before that, I want to confirm if it's more appropriate to use `{ start: boolean, end: boolean }` instead of the current option that uses enums.

**Comment by erosman** [comment]
> I'm willing to make a PR for this. Before that, I want to confirm if it's more appropriate to use `{ start: boolean, end: boolean }` instead of the current option that uses enums.

The enum was already there and new values were added to the existing options thus avoiding conflict.

https://github.com/eslint-stylistic/eslint-stylistic/blob/beef00fed4e0ba46440996e1b7fceb364a769538/packages/eslint-plugin-js/rules/padded-blocks/padded-blocks.ts#L22-L25

**Comment by 9romise** [comment]
> I'm willing to make a PR for this. Before that, I want to confirm if it's more appropriate to use `{ start: boolean, end: boolean }` instead of the current option that uses enums.

@antfu any suggestions？

**Comment by Draxare** [comment]
This feature would be really useful hoping it gets added / updated
